pkg/ccl/changefeedccl/parquet_sink_cloudstorage.go

```diff
@@ -0,0 +1,289 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package changefeedccl
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdcevent"
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvevent"
+	pqexporter "github.com/cockroachdb/cockroach/pkg/sql/importer"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/errors"
+	goparquet "github.com/fraugster/parquet-go"
+	"github.com/fraugster/parquet-go/parquetschema"
+)
+
+// This variable controls whether we add primary keys of the table to the
+// metadata of the parquet file. Currently, this will be true only under
+// testing.
+// TODO(cdc): We should consider including this metadata during production also
+var includeParquetTestMetadata = false
+
+// This is an extra column that will be added to every parquet file which tells
+// us about the type of event that generated a particular row. The types are
+// defined below.
+const parquetCrdbEventTypeColName string = "__crdb_event_type__"
```

Comment:
Looking at the remaining code -- in particular, code in the test feed to determine if parquet can be used -- I wonder if we can just support majority of the options (OptEndTime, OptResolvedTimestamps, OptDiff,
	OptMVCCTimestamps, OptUpdatedTimestamps)? 

Basically, we can do something similar to JSON encoder.
The value we store in the sentinel can be an object (or JSON blob) or array even.  The keys in the object are type (event type), updated/mvcc timestamps, etc. 

If changefeed running w/ initial_scan only -- you don't need to emit the sentinel; otherwise, emit it.  At the very least, the sentinel value will contain event type.  WDYT?